From 01c775f2e89b705d8375c0ccc0543fc82be53414 Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Tue, 7 Feb 2023 16:36:36 +0100
Subject: [PATCH] Hook up keyevents on Wayland

---
 events/xdp/xdpevents.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/events/xdp/xdpevents.cpp b/events/xdp/xdpevents.cpp
index 06c7424..ca49813 100644
--- a/events/xdp/xdpevents.cpp
+++ b/events/xdp/xdpevents.cpp
@@ -61,10 +61,8 @@ void EventData::init()
 
 void XdpEventHandler::handleKeyboard(bool down, rfbKeySym keySym)
 {
-    // TODO: implement button handling
-    // both in FakeInput interface and here
-    Q_UNUSED(down)
-    Q_UNUSED(keySym)
+    const QDBusObjectPath sessionHandle = frameBuffer()->customProperty(QStringLiteral("session_handle")).value<QDBusObjectPath>();
+    data->dbusXdpRemoteDesktopService->NotifyKeyboardKeysym(sessionHandle, {}, keySym, down);
 }
 
 void XdpEventHandler::handlePointer(int buttonMask, int x, int y)
-- 
2.39.2

